From: Andreas Mueller <amueller@nyu.edu>
Subject: add test for auto increasing

--- a/sklearn/tests/test_isotonic.py
+++ b/sklearn/tests/test_isotonic.py
@@ -186,7 +186,12 @@ def test_isotonic_regression_auto_increa
 
     # Create model and fit_transform
     ir = IsotonicRegression(increasing='auto')
-    y_ = assert_no_warnings(ir.fit_transform, x, y)
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        y_ = ir.fit_transform(x, y)
+        # work-around for pearson divide warnings in scipy <= 0.17.0
+        assert_true(all(["invalid value encountered in "
+                         in str(warn.message) for warn in w]))
 
     # Check that relationship increases
     is_increasing = y_[0] < y_[-1]
